### PR TITLE
:bug: Hide hidden rulesets from seeded hub list

### PR DESCRIPTION
**Pull request by ibolton336**
Resolves https://github.com/konveyor/tackle2-ui/issues/1093

### CODE REVIEW

File: client/src/app/queries/rulesets.ts [hunk 1]
```diff
@@ -19,25 +19,25 @@ export const useFetchRulesets = () => {
...
+              };
+
+              return {
+                ...rule,
+                metadata: transformedMetadata,
+              };
+            });
+            return { ...ruleset, rules: mappedRules };
+          })
+          .filter((ruleset) => !ruleset.name.startsWith("."));
```
Comment (ibolton336) [inline client/src/app/queries/rulesets.ts:40]: Adds this filter for hidden rulesets.

File: client/src/app/pages/migration-targets/migration-targets.tsx [hunk 2]
```diff
@@ -207,28 +203,34 @@ export const MigrationTargets: React.FC = () => {
...
-                      deleteRuleset(matchingRuleset.id);
-                    }
-                  }}
-                />
-              ))}
+              rulesetOrderSetting.data.map((id) => {
+                const matchingRuleset = rulesets.find(
+                  (Ruleset) => Ruleset.id === id
+                );
+                if (matchingRuleset) {
```
Comment (ibolton336) [inline client/src/app/pages/migration-targets/migration-targets.tsx:210]: Don't display ordered items which are hidden.

**Review by gildub — CHANGES_REQUESTED** [review]
The system owned resource is hidden in the Administration -> Custom migration targets page but there is a blank target showing in the Set Targets during Application analysis.

The concern is a succession of blank targets if there are several hidden targets:

image

**Comment by ibolton336** [comment]
> The system owned resource is hidden in the Administration -> Custom migration targets page but there is a blank target showing in the Set Targets during Application analysis.
> 
> The concern is a succession of blank targets pushing everything down in case when there are several hidden targets:
> 
> image

Good catch. The blank targets are filtered in the migration targets page, but not in the wizard.

**Review by gildub — APPROVED** [review]
LGTM

File: client/src/app/queries/rulesets.ts (hunk 1 above)
Comment (sjd78) [inline client/src/app/queries/rulesets.ts:40]: Very very minor nit-pick:  I suggest moving the filter() above the map() so the map() only needs to process things that will end up being selected.

Comment (ibolton336) [reply client/src/app/queries/rulesets.ts]: Done!

**Pull request merged**